tensorflow/contrib/seq2seq/python/kernel_tests/seq2seq_test.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Tests for contrib.seq2seq.python.seq2seq.layers_ops."""
+# pylint: disable=unused-import,g-bad-import-order
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+# pylint: enable=unused-import
+
+import tensorflow as tf
+
+
+class Seq2SeqTest(tf.test.TestCase):
+
+  # test a default call of rnn_decoder
+  def test_rnn_decoder(self):
+    pass
+
+  # test default call with time_major=True
+  def test_dynamic_rnn_decoder_time_major(self):
+    with self.test_session() as sess:
+      with tf.variable_scope("root", initializer=tf.constant_initializer(0.5)) as varscope:
+        # Define inputs/outputs to model
+        batch_size = 2
+        input_size = 3
+        decoder_input_size = 4
+        encoder_size = 8
+        decoder_size = encoder_size
+        input_sequence_length = 8
+        decoder_sequence_length = 9
+        output_size = 20
+        start_of_sequence_id = end_of_sequence_id = 3
+        decoder_embeddings = tf.get_variable('decoder_embeddings',
+            [output_size, decoder_input_size],
+            initializer=tf.random_normal_initializer(stddev=0.1))
+        inputs = tf.constant(0.5, shape=[input_sequence_length, batch_size,
+                                         input_size])
+        decoder_inputs = tf.constant(0.4, shape=[decoder_sequence_length,
+                                                 batch_size,
+                                                 decoder_input_size])
+        decoder_length = tf.constant(decoder_sequence_length, dtype=tf.int32,
+                                     shape=[batch_size,])
+
+        # Define model
+        encoder_outputs, encoder_state = tf.nn.dynamic_rnn(
+            cell=tf.nn.rnn_cell.GRUCell(encoder_size), inputs=inputs,
+            dtype=tf.float32, time_major=True, scope="rnn")
+        decoder_cell = tf.nn.rnn_cell.GRUCell(decoder_size)
+        decoder_cell = tf.nn.rnn_cell.OutputProjectionWrapper(decoder_cell,
```

Comment:
we're trying to get rid of the output projection wrapper; so let's not use it in unit tests.

i'm also confused: here you use an output projection; and in the inference one you use a separate tensor called decoder_embeddings.  when doing inference, looks like you're oging to perform a projection once and then AGAIN for decoder_embeddings.

makes more sense to just keep decoder_cell w/o this wrapper, and take the output of training and do a manual projection with decoder_embeddings.  and keep the inference_fn as is.